lib/livebook/runtime/attached.ex

```diff
@@ -47,6 +47,8 @@ defimpl Livebook.Runtime, for: Livebook.Runtime.Attached do
 
   def disconnect(runtime) do
     ErlDist.RuntimeServer.stop(runtime.server_pid)
+    _ = Node.disconnect(runtime.node)
```

Comment:
When multiple sessions are connected to the same node, we want to disconnect only once the last one is disconnected. I think with this approach we would disconnect too soon, you can verify this behaviour by starting `iex --sname test --cookie test`, connecting two sessions and disconnecting one of them.

Whenever we connect to a node we start a single `Livebook.Runtime.ErlDist.NodeManager` there and it keeps track of all connections (one `RuntimeServer` per connection). I think we could pass a `:parent_node` option when starting the node manager and then on `terminate` we would disconnect from that node. @josevalim wdyt?

Reply:
Sounds good to me!

Reply:
probably i am missunderstanding something.

If i open 2 sessions (im not sure if we are talking about 2 notebooks) and use the same node, using `Node.list/1` i still see 1 item in the list.

Maybe what we want is disconnect from the node if it's the last runtime server associated with that node?

Reply:
Under the runtime settings, you can connect to an existing node. So if you establish two different notebooks to the same node, you should only disconnect when the last node leaves. @jonatanklosko is saying there is already code to handle this case, we need to move this logic there. :)